### DISCUSSION TITLE
Connecting KPXC to browser

### DISCUSSION THREAD

**Discussion by Eynolwaden** [keepassxreboot/keepassxc#9280] [Q&A]
Hi !
I followed the steps to connect the firefox extension to the app, but when I run the script you're giving us it just cancels installation after selecting the browser. Any idea why the script can't run properly ? What can i do ?

I'm on Ubuntu last version.

Thank you all !

**Comment by droidmonkey** [comment]
Not sure what you mean, will need a screenshot.

**Reply by Eynolwaden** [reply]
So in "Browser integration" settings of  the app, KPXC tells : "Due to Snap sandboxing, you must run a script to enable browser integration. You can obtain this script from https://keepassxc.org"

I did just that, cause indeed for now the extension don't succeed to connect to KPXC.

The thing is the script doesn't succeed..

Here are the screenshots
![Screenshot from 2023-04-01 10-17-40](https://user-images.githubusercontent.com/129416844/229274696-13088682-bf13-4188-8710-9c22111be5ea.png)
![Screenshot from 2023-04-01 10-17-58](https://user-images.githubusercontent.com/129416844/229274709-e7f92599-6475-4565-a94b-0c754b7c3f24.png)
![Screenshot from 2023-04-01 10-19-01](https://user-images.githubusercontent.com/129416844/229274725-24ce8550-b742-4434-8a98-627556679735.png)

Thanks

**Reply by droidmonkey** [reply]
This script hasn't changed in a long time and it works fine for others. Make sure you are tabbing to the OK button on the first screen and pressing enter. I also recommend just switching to the flatpak installation instead, snaps are terrible.

**Reply by Eynolwaden** [reply]
Installing with flatpak I have now this in "Browser integration"...
![Screenshot from 2023-04-01 13-35-20](https://user-images.githubusercontent.com/129416844/229286429-5fb05695-b576-445f-a6f1-36f61b66878b.png)

Do I still need to run the script ?

**Reply by droidmonkey** [reply]
No need for the script

**Reply by Eynolwaden** [reply]
Well I don't understand, I select firefox (I tried with Brave also), but then in the extension clicking reload it says "Key exchange was not successful". I tried multiple times, restarting browser or app multiple times, before after etc..

**Reply by droidmonkey** [reply]
That means whatever firefox you are using is not reading configuration from the standard firefox directory. Are you using a beta version or some other off-version?

**Reply by Eynolwaden** [reply]
Oh I just saw it's written " "Browser installed as snaps are currently not supported so I'll install Brave with flatpack.. I tell you soon, thanks

**Reply by Eynolwaden** [reply]
Well doesn't work still.. I was using Firefox  that was already installed with the last version of Ubuntu.

**Reply by Eynolwaden** [reply]
Ok it now works with Firefox, I downloaded it from the official website. Thank you very much for taking the time to answer, I'm sorry I'm very new to Linux.
Enjoy your day !

**Reply by droidmonkey** [reply]
Yah Linux does nobody any favors. Too many variables and inconsistent behaviors.
